agent/fw_drupal8.c

```diff
@@ -605,6 +605,92 @@ NR_PHP_WRAPPER(nr_drupal94_invoke_all_with_clean) {
 NR_PHP_WRAPPER_END
 #endif  // OAPI
 
+/*
+ * Purpose: Instrument Drupal Attribute Hooks for Drupal 11.1+
+ *
+ * Params: 1. A zval pointer to the moduleHandler instance in use by Drupal.
+ *
+ * Return: bool
+ *
+ */
+static bool nr_drupal_hook_attribute_instrument(zval* module_handler) {
+  zval* hook_implementation_map = NULL;
+
+  nr_php_string_hash_key_t* hook_key = NULL;
+  zval* hook_val = NULL;
+  nr_php_string_hash_key_t* class_key = NULL;
+  zval* class_val = NULL;
+  nr_php_string_hash_key_t* method_key = NULL;
+  zval* module_val = NULL;
+
+  char* hookpath = NULL;
+
+  hook_implementation_map = nr_php_get_zval_object_property(
+      module_handler, "hookImplementationsMap");
+
+  if (NULL == hook_implementation_map) {
+    nrl_warning(NRL_FRAMEWORK, "NULL hookImplementationsMap object property");
+    return false;
+  }
+
+  if (!nr_php_is_zval_valid_array(hook_implementation_map)) {
+    nrl_warning(NRL_FRAMEWORK,
+                "hookImplementationsMap property not a valid array");
+    return false;
+  }
+
+  ZEND_HASH_FOREACH_STR_KEY_VAL(Z_ARRVAL_P(hook_implementation_map), hook_key,
+                                hook_val) {
+    if ((NULL == hook_key) || (0 == nr_php_is_zval_valid_array(hook_val))) {
+      nrl_warning(NRL_FRAMEWORK,
+                  "hookImplementationsMap[hook = %s]: invalid value",
```

Comment:
This can happen when `NULL == hook_key`, so more appropriate message would be:
```suggestion
                  "hookImplementationsMap[hook = %s]: invalid key or value",
```

Reply:
`value` here was intended to be generic, as in "an invalid value was returned" and not the specifc map value for a key, but I've fixed it here ea0e4cfae60efa443919a75fd33a15d1aea8bc37